static/js/topic_page/bivariate_tile.tsx

```diff
@@ -0,0 +1,332 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Component for rendering a bivariate type tile.
+ */
+
+import axios from "axios";
+import _ from "lodash";
+import React, { useEffect, useRef, useState } from "react";
+import ReactDOMServer from "react-dom/server";
+
+import { BivariateProperties, drawBivariate } from "../chart/draw_bivariate";
+import { Point } from "../chart/draw_scatter";
+import { GeoJsonData } from "../chart/types";
+import { USA_PLACE_DCID } from "../shared/constants";
+import { StatApiResponse } from "../shared/stat_types";
+import { getStatsVarLabel } from "../shared/stats_var_labels";
+import { NamedPlace } from "../shared/types";
+import { NamedTypedPlace } from "../tools/map/context";
+import { getStatsWithinPlace } from "../tools/scatter/util";
+import {
+  getNamedTypedPlace,
+  GetStatSetResponse,
+  isChildPlaceOf,
+  shouldShowMapBoundaries,
+} from "../tools/shared_util";
+import { StatVarMetadata } from "../types/stat_var";
+import { getStringOrNA } from "../utils/number_utils";
+import { getPlaceScatterData } from "../utils/scatter_data_utils";
+import { ChartTileContainer } from "./chart_tile";
+import { CHART_HEIGHT } from "./constants";
+
+interface ScatterTilePropType {
+  id: string;
+  title: string;
+  placeDcid: string;
+  enclosedPlaceType: string;
+  statVarMetadata: StatVarMetadata;
+}
+
+interface RawData {
+  geoJson: GeoJsonData;
+  placeStats: GetStatSetResponse;
+  population: StatApiResponse;
+  placeNames: { [placeDcid: string]: string };
+  place: NamedTypedPlace;
+  parentPlaces: NamedTypedPlace[];
+}
+
+interface ChartData {
+  xStatVar: {
```

Comment:
can we use StatVarMetadata here (or extract the needed part and reuse that). add a todo for that, so it can go w/ the proto change we talked about

Reply:
yes, I was planning to update that to StatVarMetadata when making the proto change. But added a todo for now